sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -5,43 +5,103 @@
 #define AZ_KEYVAULT_H
 
 #include <az_contract.h>
+#include <az_credential.h>
+#include <az_http_pipeline.h>
+#include <az_http_request_builder.h>
 #include <az_result.h>
 #include <az_span.h>
+#include <az_str.h>
 
 #include <stdlib.h>
 
 #include <_az_cfg_prefix.h>
 
-typedef struct {
-  az_span uri;
-  // TODO: Azure Credentials
-} az_keyvault_keys_client;
+extern az_span const AZ_KEY_VAULT_NONE_TYPE;
+extern az_span const AZ_KEY_VAULT_KEY_TYPE;
+extern az_span const AZ_KEY_VAULT_SECRET_TYPE;
+extern az_span const AZ_KEY_VAULT_CERTIFICATE_TYPE;
+extern az_span const AZ_KEY_VAULT_URL_PATH_SEPARATOR;
+typedef enum {
+  AZ_KEY_VAULT_NONE = 0,
+  AZ_KEY_VAULT_KEY = 1,
+  AZ_KEY_VAULT_SECRET = 2,
+  AZ_KEY_VAULT_CERTIFICATE = 3,
+} az_key_vault_key_type;
```

Comment:
I think that if the enum is named "az_key_vault_key_type", values shoud all be prefixed with AZ_KEY_VAULT_KEY_TYPE_*, i.e. AZ_KEY_VAULT_KEY_TYPE_CERTIFICATE, and  so on.

Reply:
Also, some part of me wants to change these const az_spans above from AZ_KEY_VAULT_*_TYPE to "AZ_KEY_VAULT_TYPE_*"